integrations/dynatrace/client.py

```diff
@@ -0,0 +1,104 @@
+from enum import StrEnum
+from typing import Any, AsyncGenerator
+
+import httpx
+from loguru import logger
+from port_ocean.context.event import event
+from port_ocean.utils import http_async_client
+
+
+class ResourceKey(StrEnum):
+    PROBLEM = "problem"
+    SLO = "slo"
+    ENTITY_TYPE = "entity_type"
+    ENTITY = "entity"
+
+
+class DynatraceClient:
+    def __init__(self, host_url: str, api_key: str) -> None:
+        self.host_url = host_url
+        self.client = http_async_client
+        self.client.headers = httpx.Headers({"Authorization": f"Api-Token {api_key}"})
+
+    async def _get_problems(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        url = f"{self.host_url}/api/v2/problems"
+        logger.info(f"Fetching problems from {url}")
+        while True:
```

Comment:
This block is missing try catch and logger for potential errors. We're interested in httpx.HTTPStatusError and httpx.HTTPError, in addition to other errors that may be relevant to the code.

Please update it in other functions too. If you implement the global paginated resource function, you will do it once, and apply it to all resources.

        try:
            url = f"{self.api_url}/{endpoint}"
            logger.info(
                f"URL: {url}, Method: {method}, Params: {query_params}, Body: {json_data}"
            )
            response = await self.client.request(
                method=method,
                url=url,
                params=query_params,
                json=json_data,
            )
            response.raise_for_status()

            logger.info(f"Successfully retrieved data for endpoint: {endpoint}")

            return response.json()

        except httpx.HTTPStatusError as e:
            logger.error(
                f"HTTP error on {endpoint}: {e.response.status_code} - {e.response.text}"
            )
            raise
        except httpx.HTTPError as e:
            logger.error(f"HTTP error on {endpoint}: {str(e)}")
            raise